data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -67,31 +66,52 @@ object SnapshotDeltaObject {
 
   def create[VD, ED](logs: RDD[LogTSV], snapType: SnapshotIntervalType): SnapshotDelta = {
     snapType match {
-      //case SnapshotIntervalType.Time(duration) => snapShotTime(logs, duration)
-      case SnapshotIntervalType.Count(numberOfActions) => createSnapshotCountModel(logs, numberOfActions)
+      case SnapshotIntervalType.Time(duration) => createSnapshotModel(logs.map(log => (log.timestamp.getEpochSecond, log)), duration)
+      case SnapshotIntervalType.Count(numberOfActions) => createSnapshotModel(logs.zipWithIndex().map(x => (x._2, x._1)), numberOfActions)
     }
   }
 
-  def createSnapshotCountModel(logs: RDD[LogTSV], numberOfActions: Int): SnapshotDelta = {
-    getLogger.warn(s"Creating SnapshotCountModel with numberOfActions:$numberOfActions")
-    val logsWithIndex = logs.zipWithIndex().map(x => (x._2, x._1))
-    val initialGraph = createGraph(logsWithIndex.filterByRange(0, numberOfActions).map(_._2))
+  /** Create snapshotDelta model
+   *
+   * This function works for both a timed and a counter based model. Currently the output is indistinguishable
+   * it would probably be nice to know how SnapshotDelta object was generated.
+   *
+   * It works for time as well since we assume that the timestamps have been normalized
+   * timestamps.map(timestamp => timestamp - min(timestamps))
+   *
+   * The for loop does not work for timebased applications, so making a single function maybe wasn't too smart. (but there are big similaritites)
```

Comment:
```suggestion
```
Jeg tenker det er greit å ha én funksjon. Man sender jo inn Snapshot-typen, så man er jo klar over hva som er generert. Hva refererer du til når du skriver "the for loop does not work for timebased applications"? Altså det du skrev i kommentarene lenger ned?

Reply:
Ja, at jeg er usikker på hvordan for-loopen skal funke for timebased. Så du noe på det?

Reply:
Åja, catcha ikke max() vs. count(). Tar en nærmere kikk

Reply:
Ja man sender inn snapshottypen, men SnapshotDelta-klassen burde kanskje ha et snapshotType-felt. Så man vet hvordan den ble generert